Когда используется критерий Стьюдента, а когда Z –критерий?

Z-тест (z-критерий Фишера) — класс методов статистической проверки гипотез (статистических критериев), основанных на нормальном распределении. Обычно применяется для проверки равенства средних значений при известной дисперсии генеральной совокупности или при оценке выборочного среднего стандартизованных значений. Z-статистика вычисляется как отношение разницы между случайной величиной и математическим ожиданием к стандартной ошибке этой случайной величины:

-критерий Стьюдента — общее название для класса методов статистической проверки гипотез (статистических критериев), основанных на распределении Стьюдента. Наиболее частые случаи применения t-критерия связаны с проверкой равенства средних значений в двух выборках.

t-статистика строится обычно по следующему общему принципу: в числителе — случайная величина с нулевым математическим ожиданием (при выполнении нулевой гипотезы), а в знаменателе — выборочное стандартное отклонение этой случайной величины, получаемое как квадратный корень из несмещённой оценки дисперсии.

Для применения данного критерия необходимо, чтобы выборочные средние имели нормальное распределение. При маленьких выборках это означает требование нормальности исходных значений. В случае применения двухвыборочного критерия для независимых выборок также необходимо соблюдение условия равенства дисперсий. 

In [1]:
from scipy. stats import uniform
import numpy as np
import scipy
import sympy
import scipy.stats
from scipy.stats import laplace

Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

In [2]:
a=17
sigma2=4
sigma=2
X=17.5
n=100
alpha=0.05

In [3]:
Fucrit=(1-2*alpha)/2
Fucrit

0.45

In [4]:
scipy.stats.norm.cdf(Fucrit)

0.67364477971208

In [5]:
u=((17.5-17)*np.sqrt(100))/2
u

2.5

нулевую гипотезу не принимаем

Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

In [6]:
samples=np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [7]:
n = samples.shape[0]
n

10

In [8]:
m = samples.mean()
m

198.5

In [9]:
s = samples.std(ddof=1)
s

4.453463071962462

In [10]:
print(f'n = {n},среднее = {m}, оценка среднего квадратического отклонения  = {s}')

n = 10,среднее = 198.5, оценка среднего квадратического отклонения  = 4.453463071962462


$\gamma = 0.01, k = n - 1 = 9,$ по таблице Стьюдента находим $t_y$

In [11]:
t=3.25

In [12]:
(m - t * s / np.sqrt(n), m + t * s / np.sqrt(n))


(193.9229967955732, 203.0770032044268)

утверждение продавца верно 193.9<a<203.08 данный интервал уровенем значимости 1% накрывает истинное значение измеряемой величины